pkg/kubernetes/integration_test.go

```diff
@@ -0,0 +1,308 @@
+		}
+
+		// wait for the service to be ready for accepting requests
+		err = k8s.ServiceHelper(namespace).WaitServiceReady(context.TODO(), "nginx", time.Second*20)
+		if err != nil {
+			t.Fatalf("error waiting for service nginx: %v", err)
+		}
+	})
+
+	t.Run("Exec Command", func(t *testing.T) {
+		t.Parallel()
+
+		namespace, err := createRandomTestNamespace(k8s)
+		if err != nil {
+			t.Fatal(err)
+		}
+		t.Cleanup(func() {
+			_ = k8s.Client().CoreV1().Namespaces().Delete(context.TODO(), namespace, metav1.DeleteOptions{})
+		})
+
+		busybox := fixtures.BuildBusyBoxPod()
+		_, err = k8s.Client().CoreV1().Pods(namespace).Create(
+			context.TODO(),
+			&busybox,
+			metav1.CreateOptions{},
+		)
+		if err != nil {
+			t.Fatalf("failed to create pod: %v", err)
+		}
+
+		// wait for the pod to be ready
+		running, err := k8s.PodHelper(namespace).WaitPodRunning(context.TODO(), "busybox", time.Second*20)
+		if err != nil {
+			t.Fatalf("error waiting for pod %v", err)
+		}
+		if !running {
+			t.Fatalf("timeout expired waiting for pod ready")
+		}
+
+		greetings := "hello world"
+		cmd := []string{"echo", "-n", greetings}
+		stdout, _, err := k8s.PodHelper(namespace).Exec(
+			context.TODO(),
+			"busybox",
+			"busybox",
+			cmd,
+			nil,
+		)
+		if err != nil {
+			t.Fatalf("error executing command in pod: %v", err)
+		}
+
+		if string(stdout) != greetings {
+			t.Fatalf("stdout does not match expected result:\nexpected: %s\nactual%s\n", greetings, string(stdout))
+		}
+	})
+
+	t.Run("Attach Ephemeral Container", func(t *testing.T) {
+		t.Parallel()
+
+		namespace, err := createRandomTestNamespace(k8s)
+		if err != nil {
+			t.Fatal(err)
+		}
+		t.Cleanup(func() {
+			_ = k8s.Client().CoreV1().Namespaces().Delete(context.TODO(), namespace, metav1.DeleteOptions{})
+		})
+
+		paused := fixtures.BuildPausedPod()
+		_, err = k8s.Client().CoreV1().Pods(namespace).Create(
+			context.TODO(),
+			&paused,
+			metav1.CreateOptions{},
+		)
+		if err != nil {
+			t.Fatalf("failed to create pod: %v", err)
+		}
+
+		ephemeral := corev1.EphemeralContainer{
+			EphemeralContainerCommon: corev1.EphemeralContainerCommon{
+				Name:    "ephemeral",
+				Image:   "busybox",
+				Command: []string{"sleep", "300"},
+				TTY:     true,
+				Stdin:   true,
+			},
+		}
+
+		err = k8s.PodHelper(namespace).AttachEphemeralContainer(
+			context.TODO(),
+			"paused",
+			ephemeral,
+			helpers.AttachOptions{
+				Timeout: 15 * time.Second,
+			},
+		)
+
+		if err != nil {
+			t.Fatalf("error attaching ephemeral container to pod: %v", err)
+		}
+
+		stdout, _, err := k8s.PodHelper(namespace).Exec(
+			context.TODO(),
+			"paused",
+			"ephemeral",
+			[]string{"echo", "-n", "hello", "world"},
+			nil,
+		)
+		if err != nil {
+			t.Fatalf("error executing command in pod: %v", err)
+		}
+
+		greetings := "hello world"
+		if string(stdout) != "hello world" {
+			t.Fatalf("stdout does not match expected result:\nexpected: %s\nactual%s\n", greetings, string(stdout))
```

Comment:
Same minor thing as before

```suggestion
			t.Fatalf("stdout does not match expected result:\nexpected: %q\nactual%q\n", greetings, string(stdout))
```